modules/client/src/addresslistVoting/internal/utils.ts

```diff
@@ -166,19 +166,41 @@ export function computeProposalStatus(
     parseInt(proposal.startDate) * 1000,
   );
   const endDate = new Date(parseInt(proposal.endDate) * 1000);
+  // The proposal is executed so the status becomes EXECUTED
+  // independently of the other conditions
   if (proposal.executed) {
     return ProposalStatus.EXECUTED;
   }
+  // The proposal is not executed and the start date is in the future
+  // so the status becomes PENDING
   if (startDate >= now) {
     return ProposalStatus.PENDING;
   }
-  if (proposal.potentiallyExecutable || proposal.earlyExecutable) {
-    return ProposalStatus.SUCCEEDED;
+  // The proposal is not executed and the start date is in the past
+  // So we must check if the proposal reached the approval threshold
+  // If it reached the approval threshold and it's a signaling proposal
+  // the status becomes SUCCEEDED
+  // If it reached the approval threshold and it's not a signaling proposal
+  // the status becomes SUCCEEDED if the end date is in the past or if the
+  // proposal is early executable
+  if (proposal.approvalReached) {
+    if(proposal.isSignaling) {
+      return ProposalStatus.SUCCEEDED;
+    } else {
+      if (now >= endDate || proposal.earlyExecutable) {
+        return ProposalStatus.SUCCEEDED;
+      }
+    }
   }
```

Comment:
I'm not sure if this is correect, this would mark as succeeded a proposal that hasnt ended and maybe is not early executable, so the UI would mark it as executable when is not the case

Reply:
> this would mark as succeeded a proposal that hasnt ended

Exactly, this is the idea behind the early execution feature.

> maybe is not early executable

`proposal.earlyExecutable` should only be set to true if it is early executable. Here are [the docs on how `earlyExecution` mode works](https://devs.aragon.org/docs/osx/reference-guide/plugins/governance/majority-voting/MajorityVotingBase#early-execution).